lnwallet/btcwallet/signer.go

```diff
@@ -39,13 +37,40 @@ func (b *BtcWallet) FetchInputInfo(prevOut *wire.OutPoint) (*wire.TxOut, error)
 	// we actually have control of this output. We do this because the check
 	// above only guarantees that the transaction is somehow relevant to us,
 	// like in the event of us being the sender of the transaction.
-	output = txDetail.TxRecord.MsgTx.TxOut[prevOut.Index]
-	if _, err := b.fetchOutputAddr(output.PkScript); err != nil {
+	pkScript := txDetail.TxRecord.MsgTx.TxOut[prevOut.Index].PkScript
+	if _, err := b.fetchOutputAddr(pkScript); err != nil {
 		return nil, err
 	}
 
+	// Then, we'll populate all of the information required by the struct.
+	addressType := lnwallet.UnknownAddressType
+	switch {
+	case txscript.IsPayToWitnessPubKeyHash(pkScript):
+		addressType = lnwallet.WitnessPubKey
+	case txscript.IsPayToScriptHash(pkScript):
+		addressType = lnwallet.NestedWitnessPubKey
+	}
+
+	// Determine the number of confirmations the output currently has.
+	_, currentHeight, err := b.GetBestBlock()
+	if err != nil {
+		return nil, fmt.Errorf("unable to retrieve current height: %v",
+			err)
+	}
+	confs := int64(0)
+	if txDetail.Block.Height != -1 {
```

Comment:
`-1` means unconfirmed?

Reply:
Yeah that's what's returned by the underlying call for unconfirmed transactions.

Reply:
`const unconfirmed = -1` would be nice